src/v/kafka/server/group_router.cc

```diff
@@ -16,6 +16,154 @@
 
 namespace kafka {
 
+template<typename Request, typename FwdFunc>
```

Comment:
Note that this PR has two changes: the first change simply moves most of the implementation of `group_router` into the `.cc` file to allow 30-second iteration instead of 10 minutes, while the substance of the change is in the second commit. It is probably easiest to look at the 1st -> 2nd commit diff to see those changes alone.